source/client-side-encryption/client-side-encryption.rst

```diff
@@ -862,24 +901,75 @@ Drivers MUST support a BSON document option named ``encryptedFields`` for any
 will be interpreted by the helper method and MUST be passed to the `create`_
 command.
 
-A call to a driver helper ``CreateCollection(collectionName, collectionOptions)`` must check if the collection namespace (``<databaseName>.<collectionName>``) has an associated ``encryptedFields``. Check for an associated ``encryptedFields`` from the following:
-
-- The ``encryptedFields`` option passed in ``collectionOptions``.
-- The value of ``AutoEncryptionOpts.encryptedFieldsMap[<databaseName>.<collectionName>]``.
+For a helper function, ``CreateCollection(collectionName, collectionOptions)``
+with the name of the database associated as `dbName`, look up the encrypted
+fields ``encryptedFields`` for the collection as
+`GetEncryptedFields(collectionOptions, collectionName, dbName, false)`
+(`See here <GetEncryptedFields_>`_).
+
+If a set of ``encryptedFields`` was found, then do the following operations. If
+any of the following operations error, the remaining operations are not
+attempted:
+
+- Create the collection with name ``encryptedFields["escCollection"]`` as a
+  clustered collection using the options
+  ``{clusteredIndex: {key: {_id: 1}, unique: true}}``. If
+  ``encryptedFields["escCollection"]`` is not set, use the collection name
+  ``enxcol_.<collectionName>.esc``. Creating this collection MUST NOT check if
+  the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection with name ``encryptedFields["eccCollection"]`` as a
+  clustered collection using the options
+  ``{clusteredIndex: {key: {_id: 1}, unique: true}}``. If
+  ``encryptedFields["eccCollection"]`` is not set, use the collection name
+  ``enxcol_.<collectionName>.ecc``. Creating this collection MUST NOT check if
+  the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection with name ``encryptedFields["ecocCollection"]`` as a
+  clustered collection using the options
+  ``{clusteredIndex: {key: {_id: 1}, unique: true}}``. If
+  ``encryptedFields["ecocCollection"]`` is not set, use the collection name
+  ``enxcol_.<collectionName>.ecoc``. Creating this collection MUST NOT check if
+  the collection namespace is in the ``AutoEncryptionOpts.encryptedFieldsMap``.
+- Create the collection ``collectionName`` with ``collectionOptions`` and the
+  option ``encryptedFields`` set to the ``encryptedFields``.
+- Create the the index ``{"__safeContent__": 1}`` on collection
+  ``collectionName``.
+
+
+Create Encrypted Collection Helper
```

Comment:
I realize we don't have a [Collection Management spec](https://github.com/mongodb/specifications/tree/master/source/collection-management/tests) (just tests), but can a DRIVERS ticket be created and linked to DRIVERS-2312 to serve as a reminder to reference this helper when such a spec is ultimately created?

Reply:
![](https://media.giphy.com/media/xjmzeEXiaTTk7WpeCm/giphy.gif)

https://jira.mongodb.org/browse/DRIVERS-2481